### ISSUE TITLE
`TypeForm` doesn't work with type expression that includes a traditional TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#9159]
If a `TypeVar` is allocated using the traditional (pre-PEP 695) manner, a type expression that uses this type variable isn't considered a valid type form.

```python
from typing import TypeVar
from typing_extensions import TypeForm

T = TypeVar("T")

def func(x: T) -> T:
    v: TypeForm[T] = T # Should not generate error
    return x
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.384
